shell/platform/darwin/ios/framework/Source/FlutterPluginAppLifeCycleDelegate.mm

```diff
@@ -37,7 +37,7 @@ - (void)addObserverFor:(NSString*)name selector:(SEL)selector {
   dispatch_block_t unsubscribe = ^{
     [[NSNotificationCenter defaultCenter] removeObserver:blockSelf name:name object:nil];
```

Comment:
To @stuartmorgan's point, this looks like a new retain cycle.  This object strongly retains _notificationUnsubscribers, which strongly retains blocks that strongly retain this object.

- Naive fix is to weakly retain self in the block.
- Better fix would be to convert _notificationUnsubscribers to a NSMutableArray<NSString *> of notification NAMES to remove the observe to in dealloc.
- Even better would be to just call [[NSNotificationCenter defaultCenter] removeObserver:self] in -dealloc which handles all notifications (though not needed at all in iOS 9.0 or macOS 10.11 and later).